File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        from itertools import combinations
        ans = 0
        combos = []
        for i in range(1, len(nums)+1):
            combos += combinations(nums, i)
        print(combos)
        for combo in combos:
            base = combo[0]
            for x in combo[1:]:
                base ^= x
            ans += base
        return ans

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        return XORSum(nums, 0, 0);    
    }
private:
    int XORSum(vector<int>& nums, int index, int currSum) {
        if (index == nums.size()) return currSum;

        int withCurrent = XORSum(nums, index+1, currSum^nums[index]);
        int withoutCurrent = XORSum(nums, index+1, currSum);

        return withCurrent + withoutCurrent;
    }
};